app/TuistApp/Sources/Views/DevicesView/DevicesViewModel.swift

```diff
@@ -56,22 +56,43 @@ struct PinnedSimulatorsKey: AppStorageKey {
 
 struct SelectedDeviceKey: AppStorageKey {
     static let key = "selectedSimulator"
-    static let defaultValue: SelectedDevice? = nil
+    static let defaultValue: ReducedDevice? = nil
 }
 
 enum SelectedDevice: Codable, Equatable {
     case simulator(SimulatorDeviceAndRuntime)
     case device(PhysicalDevice)
 }
 
+enum ReducedDevice: Codable, Equatable {
+    case simulator(id: String)
+    case device(id: String)
```

Comment:
I’ve defined a new structure to store the selected state. Do you have any suggestions for improvement?